### PR TITLE
feat: user-friendly error messages

### LINKED ISSUE #144: User friendly error messages
The sidekick currently displays a generic "failed to fetch staus" message when something goes wrong. The user is left to guess what the underlying cause might be. Ideally the sidekick would reveal some hints how to fix it.

### PR DESCRIPTION
**Pull request by rofe**
Fix #144 

Checks for 401 and 404 errors and gives user some hints.

### CODE REVIEW

File: src/sidekick/module.js [hunk 1]
```diff
@@ -960,18 +960,44 @@
...
+          // check for error status
+          if (!resp.ok) {
+            let msg = '';
+            switch (resp.status) {
+              case 401:
+                msg = 'Not authorized to access page. Check your login or network status.';
+                break;
+              case 404:
+                msg = this.isEditor()
+                  ? 'Page not found. Check Helix access to this document or Sidekick configuration.'
```
Comment (kptdobe) [inline src/sidekick/module.js:973]: It is maybe me but I had to read this one multiple times to understand what it means.

File: src/sidekick/module.js [hunk 2]
```diff
@@ -960,18 +960,44 @@
...
+                  : 'Page not found. Check your URL or Sidekick configuration.';
+                break;
+              default:
+                msg = 'Failed to fetch status. Please try again later.';
```
Comment (kptdobe) [inline src/sidekick/module.js:977]:
```suggestion
                msg = 'Failed to fetch the page status. Please try again later.';
```

File: src/sidekick/module.js (hunk 1 above)
Comment (kptdobe) [reply src/sidekick/module.js:973]:
```suggestion
                  ? 'Page not found. Check if Helix has access to this document or your Sidekick configuration.'
```

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: src/sidekick/module.js (hunk 1 above)
Comment (rofe) [reply src/sidekick/module.js:973]: Thanks! Tried to keep it (too) short 😄

**Review by davidnuescheler — APPROVED** [review]

**Comment by davidnuescheler** [comment]
🥳

**Review by kptdobe — APPROVED** [review]

**Pull request merged**